game/ui/inventorymenu.cpp

```diff
@@ -145,6 +145,9 @@ bool InventoryMenu::ransack(Npc &pl, Npc &tr) {
   auto it = tr.inventory().iterator(Inventory::T_Ransack);
   if(!it.isValid())
     return false;
+  // fix quest "Lost in Darkness"
+  if (tr.displayName()=="Sengrath" && tr.isDead())
+    tr.addItem(7674,1);
```

Comment:
I'm not really Gothic expert but this looks suspicious at best. What if there is a character named Sengrath in some mod, did the original game have random quest specific code scattered in the engine code?

To me it sounds like there is some sort of bug in handling of this specific quest, that should be fixed rather than hardcoding a workaround into the engine.

I might be completely wrong tho, I'm just a random bystander.